book/chapters/chapter10/advanced_technical_aspects_of_mlr3.qmd

```diff
@@ -78,12 +75,17 @@ system.time({parSapply(cl, x, f, chunk.size = 1)})
 system.time({parSapply(cl, x, f, chunk.size = 2500)})
 ```
 
-Whenever you have the option to control the granularity by setting the chunk size, you should aim for at least as many jobs as workers and also the runtime of each worker should be at least several seconds.
-This ensures that you can fully utilize the system and that the parallelization overhead stays reasonable.
+Whenever you have the option to control the granularity by setting the chunk size, you should aim for at least as many jobs as workers.
+However, if there are too few job chunks with strongly dissimilar runtimes, the system may end up waiting for the last chunk to finish, while other ressources are idle.
+This is referred to as `r index("synchronization overhead", aside = TRUE)`.
+You should therefore aim for chunks with a runtime of at least several seconds, so that the parallelization overhead remains reasonable, while still having enough chunks to ensures that you can fully utilize the system.
+
 If you have heterogeneous runtimes, you can consider grouping jobs so that the runtimes of the chunks are more homogeneous.
 If runtimes can be estimated, then `r ref("batchtools::binpack()")` and `r ref('batchtools::lpt()')` are useful for chunking jobs.
-If runtimes cannot be estimated, then it can be useful to randomize the order of jobs, this can prevent long jobs all being executed at the end, which leads to avoidable underutilization of resources.
-`r mlr3misc` ships with the functions `r ref("chunk()")` and `r ref("chunk_vector()")` to conveniently chunk jobs and also shuffles them per default.
+If runtimes cannot be estimated, then it can be useful to randomize the order of jobs.
+It could otherwise happen that jobs are accidentally ordered by runtime, e.g. because they are sorted by a hyperparameter that has a strong influence on training time.
```

Comment:
```suggestion
Otherwise jobs could be accidentally ordered by runtime, for example because they are sorted by a hyperparameter that has a strong influence on training time.
```